### ISSUE TITLE
Pure keyword in submodule interface definitions in fortran code leads to error

### ISSUE THREAD

**Issue by mscfd** [ninja-build/ninja#1594]
Fortran code in a big code base which is similar to the one below, leads to an error in dependency analysis. I tested with the latest kitware branch as well as master (from what I can see, fortran is completely merged by now). The problem looks very much like an oversight in the fortran parser for dependency analysis.
ninja aborts with the wrong error message:
`ninja: build stopped: multiple rules generate mod/pure.mod`
because it thinks that pure is a module (due to the 'module pure' prefix).
There are other keywords which should cause the same error (elemental, impure, ...?)

(Remark: I have not tested the code below itself, as I do not have setup a cmake/ninja build system for this small example).

mod.f90:
```
module mod
implicit none

interface xxx
   module pure function xxx(i) result(j)
      integer :: j
      integer, intent(in) :: i
   end function xxx
end interface xxx

interface yyy
   module pure function yyy(i) result(j)
      integer :: j
      integer, intent(in) :: i
   end function yyy
end interface yyy
end module mod
```

mod_smod.f90:
```
submodule (mod) mod_smod

implicit none

contains

module pure function xxx(i) result(j)
      integer :: j
      integer, intent(in) :: i
      j = i + 1
end function xxx

module pure function yyy(i) result(j)
      integer :: j
      integer, intent(in) :: i
      j = i + 1
end function yyy

end submodule mod_smod
```

**Comment by scivision** [comment]
Hi @mscfd , this is a long-standing issue in CMake itself that I have experienced over time, and it occurs for me with Cmake + GNU Make as well.

 Meson + Ninja does not have this Fortran submodule issue (I fixed this issue in Meson).  Not to humblebrag or get you to switch build systems ;-)

**Comment by mscfd** [comment]
@scivision: Thanks for pointing out cmake. For a couple of reasons I thought that ninja was to blame here.
For anybody caring about this, here is the issue within cmake
https://gitlab.kitware.com/cmake/cmake/issues/18427

Unfortunately I cannot change the build system. On the other hand, I got the feeling that submodule support is still a bit buggy (in particular with gfortran), as I ran into a number of issues. With ifort I only had one kind of funny problem as it did not accept a module subroutine called "get", any other name was eligible (and there was no other entity called get anywhere near).
